src/profiler/profiler.c

```diff
@@ -237,20 +270,22 @@ void xdebug_profiler_deinit()
 		xdebug_profiler_function_end(fse);
 	}
 
-	fprintf(
-		XG_PROF(profile_file),
+	profiler_fprintf(
 		"summary: %lu %zd\n\n",
 		(unsigned long) ((xdebug_get_utime() - (XG_PROF(profiler_start_time))) * 1000000),
 		zend_memory_peak_usage(0)
 	);
 
 	XG_PROF(active) = 0;
 
-	fflush(XG_PROF(profile_file));
+	profiler_fflush(1);
```

Comment:
I prefer having constants instead of arbitrary numbers (`0` and `1`). The name is also misnomer, as it's not actually always flushing. It just writes it to the file, and doesn't flush the OS buffer with ``fflush`` always.